invoice/views.py

```diff
@@ -71,17 +51,68 @@ class BankAccountListView(ListView):
 
 class CustomerCreateView(CreateView):
     """Create a new customer."""
+    template_name = 'invoice/customer_form.html'
     model = Customer
-    fields = '__all__'
+    fields = ['first_name', 'last_name', 'email']
     success_url = reverse_lazy('customer-list')
 
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        if self.request.POST:
+            context['address_form'] = AddressForm(self.request.POST)
+        else:
+            context['address_form'] = AddressForm()
+        return context
+
+    def form_valid(self, form):
+        """Create a new customer and a new address."""
+        address_form = AddressForm(self.request.POST)
+        address = address_form.save()
+        customer = form.save(commit=False)
+        customer.address = address
+        return super().form_valid(form)
 
-class CustomerUpdateView(UpdateView):
+
+class CustomerUpdateView(FormView):
     """Update an existing customer."""
-    model = Customer
-    fields = '__all__'
+    template_name = 'invoice/customer_form.html'
+    form_class = CustomerForm
     success_url = reverse_lazy('customer-list')
 
+    def get_context_data(self, **kwargs):
+        customer = Customer.objects.get(id=self.kwargs['pk'])
+        context = super().get_context_data(**kwargs)
+        form_data = customer.dict()
+        context['form'] = CustomerForm(initial=form_data)
+        return context
+
+    def form_valid(self, form):
+        """Create a new customer and a new address."""
+        customer_id = self.kwargs['pk']
+        address_line_1 = form.cleaned_data['address_line_1']
+        address_line_2 = form.cleaned_data['address_line_2']
+        address_line_3 = form.cleaned_data['address_line_3']
+        address_postcode = form.cleaned_data['address_postcode']
+        address_city = form.cleaned_data['address_city']
+        address_state = form.cleaned_data['address_state']
+        address_country = form.cleaned_data['address_country']
+        customer = Customer.objects.get(id=customer_id)
+        customer.first_name = form.cleaned_data['first_name']
+        customer.last_name = form.cleaned_data['last_name']
+        customer.email = form.cleaned_data['email']
+        address = customer.address
+        address.line_1 = address_line_1
+        address.line_2 = address_line_2
+        address.line_3 = address_line_3
+        address.postcode = address_postcode
+        address.city = address_city
+        address.state = address_state
+        address.country = address_country
+        address.save()
+        customer.save()
```

Comment:
is there a cleaner way to do this instead of manually querying all the fields?
cant you use the modelform here?

Reply:
Indeed, wasn't done, yet